app/src/main/java/com/metro/gwuexplorer/MainActivity.kt

```diff
@@ -44,37 +43,17 @@ class MainActivity : AppCompatActivity() {
             .show()
 
 
-
-        if (checkedBox.isChecked()) {
-
-            // Pass the name and the file-create mode (e.g. private to our app)
-            val preferences = getSharedPreferences("gwu-explorer", Context.MODE_PRIVATE)
-            // Writing to preferences (make sure you call apply)
-            preferences.edit().putString("saved_stationName", stationname.text.toString()).apply()
-            // Reading from preferences, indicate default if not present
-            val savedStationName = preferences.getString("saved_stationName", "")
-
-        }
-
-
-        
         go.setOnClickListener {
 
-           // val choices = listOf("667 M Street Washington, DC 20585", "668 M Street Washington, DC 20585")
-
-
-
                 // Pass a context (e.g. Activity) and locale
                 val geocoder = Geocoder(this, Locale.getDefault())
-                val locationName = stationname.getText().toString()
-
+                val locationName : String? = stationname.getText().toString()
                 val maxResults = 3
+                val results: List<Address>? = geocoder.getFromLocationName(locationName, maxResults)
```

Comment:
I don't think this variable needs to be nullable (e.g. can just be `List<Address>`. If there are no results, the geocoder will return an empty list